prime-router/src/main/kotlin/azure/ActionHistory.kt

```diff
@@ -572,7 +602,25 @@ class ActionHistory {
         filteredOutReports.values.forEach { it.actionId = action.actionId }
         insertReports(txn)
         DatabaseAccess.saveTestData(covidMetaDataRecords, txn)
-        generateReportLineages(action.actionId)
+
+        // TODO: Generation of lineages needs to be separated from database insert functionality, since there
+        //  are conditional lineage generation rules
+        if (generatingEmptyReport) {
+            // if we are generating an empty report for the 'send' step there will be one report in and one out.
+            //  make sure to track the lineage. for the 'batch 'step there will not be any lineage
+            if (reportsIn.size == 1 && reportsOut.size == 1)
```

Comment:
If a empty report is being generated and this is not true, an error should be thrown.

Reply:
This is hit twice for one 'empty batch' process - once from BatchFunction which will have no reportIn and therefore does not need lineage, and once during SendFunction which will have both a reportsIn and a reportsOut, requiring lineage.